src/lib/Microsoft.Health.Extensions.Fhir.R4/Service/ResourceManagementService.cs

```diff
@@ -71,5 +83,16 @@ private static Model.Identifier BuildIdentifier(string value, string system)
             var identifier = new Model.Identifier { Value = value, System = string.IsNullOrWhiteSpace(system) ? null : system };
             return identifier;
         }
+
+        private static Guid ToGuid(string value)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(value, nameof(value));
+
+            using (MD5 md5 = MD5.Create())
+            {
+                byte[] hash = md5.ComputeHash(Encoding.Default.GetBytes(value));
+                return new Guid(hash);

```

Comment:
That would be an improvement.  I assume we are losing some of that when we convert it into a GUID.  What about just doing a base64 encoding?

Reply:
Yeah base64 seems safer from a collision avoiding standpoint.

There was one small problem with base64 which is that it can output '+' and '/' and '=' which would mess up the query params and prevent the resource from being saved, so I replaced those characters with some other characters in an attempt to be consistent with RFC 4648: https://en.wikipedia.org/wiki/Base64#The_URL_applications